src/cc/api/BPF.cc

```diff
@@ -55,28 +55,44 @@ std::string sanitize_str(std::string str, bool (*validator)(char),
   return str;
 }
 
+StatusTuple BPF::init_usdt(const USDT& usdt) {
+  usdt_.emplace_back(usdt);
+  StatusTuple init_stp = usdt_.back().init();
+  if (init_stp.code() != 0) {
+    usdt_.pop_back();
```

Comment:
is there some internal mutability inside USDT? If not, it seems weird to

1. emplace usdt
2. use back() to get it back
3. do init
4. if it fails - pop from vector

Instead of:
1. usdt.init()
2. if it succeeds - add to usdt_

Thoughts?

Reply:
i thought there was some weirdness with the move constructor when the usdt had already been `init`ted, but that is not the case. Agreed with your suggestions and resolved.